### PR TITLE
VP-5985: Add sort by scoring relevance

### PR DESCRIPTION
**Pull request by AliveMen**
### Related task:    
VP-5985 Sort by scoring relevance and test all sorting scenarios
### Changes: 
Add sorting by relevance

### CODE REVIEW

File: src/VirtoCommerce.ElasticSearchModule.Data/ElasticSearchRequestBuilder.cs [hunk 1]
```diff
@@ -9,16 +9,21 @@ namespace VirtoCommerce.ElasticSearchModule.Data
...
             var result = new Nest.SearchRequest(indexName)
             {
                 Query = GetQuery(request),
                 PostFilter = GetFilters(request, availableFields),
                 Aggregations = GetAggregations(request, availableFields),
                 Sort = GetSorting(request?.Sorting),
                 From = request?.Skip,
-                Size = request?.Take
+                Size = request?.Take,
+                TrackScores = request?.Sorting.Any(x=>x.FieldName == Score) ?? false
```
Comment (yecli) [inline src/VirtoCommerce.ElasticSearchModule.Data/ElasticSearchRequestBuilder.cs:26]: Could you use our extension `string.EqualsInvariant` to compare strings

File: src/VirtoCommerce.ElasticSearchModule.Data/ElasticSearchRequestBuilder.cs [hunk 2]
```diff
@@ -93,6 +98,14 @@ protected virtual ISort GetSortingField(SortingField field)
                     Order = geoSorting.IsDescending ? SortOrder.Descending : SortOrder.Ascending,
                 };
             }
+            else if (field.FieldName.ToLowerInvariant() == Score)
```
Comment (yecli) [inline src/VirtoCommerce.ElasticSearchModule.Data/ElasticSearchRequestBuilder.cs:101]: Could you use our extension `string.EqualsInvariant` to compare strings

File: src/VirtoCommerce.ElasticSearchModule.Data/ElasticSearchRequestBuilder.cs (hunk 1 above)
Comment (AliveMen) [reply src/VirtoCommerce.ElasticSearchModule.Data/ElasticSearchRequestBuilder.cs]: Fixed

File: src/VirtoCommerce.ElasticSearchModule.Data/ElasticSearchRequestBuilder.cs (hunk 2 above)
Comment (AliveMen) [reply src/VirtoCommerce.ElasticSearchModule.Data/ElasticSearchRequestBuilder.cs]: Fixed

**Review by yecli — DISMISSED** [review]

**Review by Vectorfield4 — APPROVED** [review]

**Review by tatarincev — APPROVED** [review]

**Review by yecli — APPROVED** [review]

**Pull request merged**